### PR TITLE
Add e2e test for merchant orders > apply coupon

### LINKED ISSUE #28359: Write new e2e test: Merchant / Orders / Apply Coupon
Write a new e2e test that covers `Merchant / Orders / Apply Coupon` flow.

### PR DESCRIPTION
**Pull request by zhongruige**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR adds a new suite of e2e tests covering the `Merchant / Orders / Apply Coupon` flow, testing applying a coupon to an order. The run time is around less than a second for the test:

Screen Shot 2020-12-15 at 8 11 30 PM

Additionally, the following helper methods was added:
- `createCoupon( couponAmount )`: creates a basic coupon with the provided coupon amount (defaults to 5). Returns the generated coupon code.

Closes https://github.com/woocommerce/woocommerce/issues/28359.

### How to test the changes in this Pull Request:

1. Make sure the new e2e tests pass locally and on Travis. The tests can be ran following [these instructions](https://github.com/woocommerce/woocommerce/tree/master/tests/e2e#running-tests).

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

N/A

### CODE REVIEW

File: tests/e2e/utils/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
 - `moveAllItemsToTrash()` util helper method that checks every item in a list page and moves them to the trash
 - `createSimpleOrder( status )` component which accepts an order status string and creates a basic order with that status
 - `addProductToOrder( orderId, productName )` component which adds the provided productName to the passed in orderId
+- `createCoupon( couponAmount )` component which accepts a cooupon amount string (it defaults to 5) and creates a basic coupon. Returns the generated coupon code.
```
Comment (tammullen) [inline tests/e2e/utils/CHANGELOG.md:13]: Small typo on `accepts a cooupon`

Comment (zhongruige) [reply tests/e2e/utils/CHANGELOG.md]: Good catch--fixed!

File: tests/e2e/core-tests/specs/merchant/wp-admin-order-apply-coupon.test.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+			await uiUnblocked();
+
+			// Verify the coupon list is showing
+			await page.waitForSelector('.wc-used-coupons');
+			await expect(page).toMatchElement('.wc_coupon_list', { text: 'Coupon(s)' });
+			await expect(page).toMatchElement('.wc_coupon_list li.code.editable', { text: couponCode });
+
+			// Check that the coupon has been applied
+			await expect(page).toMatchElement('.wc-order-item-discount', { text: '5.00' });
+			await expect(page).toMatchElement('.line_cost > .view > .woocommerce-Price-amount', { text: '4.99' });
```
Comment (tammullen) [inline tests/e2e/core-tests/specs/merchant/wp-admin-order-apply-coupon.test.js:57]: I think this is fine for this PR but maybe we can do something similar (or as part of it)to https://github.com/woocommerce/woocommerce/pull/28575 for these card coded values later.

**Review by tammullen — APPROVED** [review]
Tests work great in headless! Thanks for this. Filter Orders by Status fail in non headless but that is unrelated to this PR.

File: tests/e2e/core-tests/specs/merchant/wp-admin-order-apply-coupon.test.js (hunk 2 above)
Comment (zhongruige) [reply tests/e2e/core-tests/specs/merchant/wp-admin-order-apply-coupon.test.js:57]: Agreed! I was thinking I would circle back on that PR and update these tests to use the new config variable once they're in `master`.

**Pull request merged**